vortex-layout/src/layouts/dict/writer/mod.rs

```
@@ -99,8 +99,8 @@ impl LayoutStrategy for DictStrategy {
             })
             .await?;
             let stream = SequentialStreamAdapter::new(dtype.clone(), stream).sendable();
-            if !is_dict_encoding? {
-                // first chunk did not compress to dict, skip dict layout
+            if !is_dict_encoding.unwrap_or(false) {
```

I think we are swallowing errors here

---

yea you are right

---

looks better now I think, should have done it like this the first time. Only if Stream::peekable was easier to use here